### PR TITLE
Remove the storage key requirement for linked cluster with introduced livy interactive APIs

### PR DESCRIPTION
**Pull request by wezhang**
Signed-off-by: Wei Zhang <wezhang@outlook.com>

### CODE REVIEW

**Comment by wezhang** [comment]
Compiling timeout:

```
Still running (20 of 20): ./BuildDevint -q
Timeout (20 minutes) reached. Terminating "./BuildDevint -q"
The command ./BuildDevint -q exited with 137.
```

**Comment by wezhang** [comment]
Travis is slower than Jenkins:  `Ran for 26 min 4 sec`, Jenkins only took 6mins.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java [hunk 1]
```diff
@@ -119,8 +97,47 @@ protected Control createDialogArea(Composite parent) {
...
+        gridData = new GridData();
+        gridData.horizontalAlignment = SWT.FILL;
+        gridData.grabExcessHorizontalSpace = true;
+        passwordField.setLayoutData(gridData);
+        passwordField.setToolTipText("The password of the HDInsight cluster user provided.\n\n Press the F1 key or click the '?'(Help) button to get more details.");
+
         Group clusterStorageGroup = new Group(container, SWT.NONE);
-        clusterStorageGroup.setText("The Cluster Storage Information");
+        clusterStorageGroup.setText("The Cluster Storage Information (Optional)");
+        clusterStorageGroup.setToolTipText("The Cluster Storage Information provided can enable the Storage Explorer support.");
```
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java]: >Storage Explorer support. [](start = 100, length = 25)

The Cluster Storage Information provided can enable the Storage Explorer support in the left tree view.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/rxjava/IdeaSchedulers.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                public void run(@NotNull ProgressIndicator progressIndicator) {
+                    command.run();
+                }
+            };
+
+            final ProgressIndicator progressIndicator = new BackgroundableProcessIndicator(task);
+
+            ProgressManager.getInstance().runProcessWithProgressAsynchronously(task, progressIndicator);
+        }));
+    }
```
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/rxjava/IdeaSchedulers.java]: These two functions look the same

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/rxjava/EclipseSchedulers.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.swt.widgets.Display;
+
+import com.microsoft.azure.hdinsight.common.mvc.IdeSchedulers;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+
+import rx.Scheduler;
+import rx.schedulers.Schedulers;
+
+public class EclipseSchedulers implements IdeSchedulers {
```
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/rxjava/EclipseSchedulers.java]: >EclipseSchedulers [](start = 13, length = 17)

Assum there will be IntelliJSchedulers?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/rxjava/IdeaSchedulers.java (hunk 2 above)
Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/rxjava/IdeaSchedulers.java]: Almost, the one above is `invokeLater`, but this one is `invokeAndWait`

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java (hunk 1 above)
Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java]: Fix it.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/rxjava/EclipseSchedulers.java (hunk 3 above)
Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/rxjava/EclipseSchedulers.java]: Yes, the one for IntelliJ is `IdeaSchedulers`

**Comment by wezhang** [comment]
to fix #1536

**Pull request merged**